lib/osut/utils.rb

```diff
@@ -5128,11 +5275,13 @@ def daylit?(space = nil, sidelit = true, toplit = true, baselit = true)
   # @option subs [#to_f] :r_buffer gap between sub/array and right corner
   # @option subs [#to_f] :l_buffer gap between sub/array and left corner
   # @param clear [Bool] whether to remove current sub surfaces
+  # @param bound [Bool] whether to add subs wrt surface's bounded box
+  # @param realign [Bool] whether to first realign bounded box
   # @param bfr [#to_f] safety buffer, to maintain near other edges
   #
   # @return [Bool] whether addition is successful
   # @return [false] if invalid input (see logs)
-  def addSubs(s = nil, subs = [], clear = false, bfr = 0.005)
+  def addSubs(s = nil, subs = [], clear = false, bound = false, realign = false, bfr = 0.005)
```

Comment:
Setting options _bound_ and _realign_ to **true** became key to the systematic treatment of roofs (as candidates for skylights), whether flat, low sloped, etc.